core/src/main/java/jenkins/slaves/DefaultJnlpSlaveReceiver.java

```diff
@@ -149,8 +149,9 @@ public void beforeChannel(@NonNull JnlpConnectionState event) {
         final SlaveComputer computer = state.getNode();
         final OutputStream log = computer.openLogFile();
         state.setLog(log);
-        PrintWriter logw = new PrintWriter(log, true);
-        logw.println("Inbound agent connected from " + event.getSocket().getInetAddress());
+        try (PrintWriter logw = new PrintWriter(log, true)) {
+            logw.println("Inbound agent connected from " + event.getSocket().getInetAddress());
+        }
```

Comment:
AFAICT it would also close `log`, and it is not supposed to happen here

Reply:
reverted here also.